darts/timeseries.py

```diff
@@ -140,156 +582,251 @@ def pd_dataframe(self, copy=True) -> pd.DataFrame:
+            The first values of every component of this deterministic time series
         """
-        return self._values[0]
+        self._assert_deterministic()
+        return self._xa.values[0, :, 0].copy()
 
     def last_values(self) -> np.ndarray:
         """
         Returns
         -------
         np.ndarray
-            The last values of every component of this time series
+            The last values of every component of this deterministic time series
         """
-        return self._values[-1]
+        self._assert_deterministic()
+        return self._xa.values[-1, :, 0].copy()
 
-    def values(self, copy=True) -> np.ndarray:
+    def values(self, copy=True, sample=0) -> np.ndarray:
         """
+        Returns a 2-D Numpy array of dimension (time, component), containing this series' values for one sample.
+        If this series is deterministic, it contains only one sample and only `sample=0` can be used.
+
         Parameters
         ----------
         copy
-            Whether to return a copy of the values. Leave it to True unless you know what you are doing.
+            Whether to return a copy of the values, otherwise returns a view.
+            Leave it to True unless you know what you are doing.
 
         Returns
         -------
         numpy.ndarray
             The values composing the time series.
         """
+        raise_if(self.is_deterministic and sample != 0, 'This series contains one sample only (deterministic),'
+                                                        'so only sample=0 is accepted.', logger)
         if copy:
-            return np.copy(self._values)
+            return np.copy(self._xa.values[:, :, sample])
         else:
-            return self._values
+            return self._xa.values[:, :, sample]
 
-    def univariate_values(self, copy=True) -> np.ndarray:
+    def all_values(self, copy=True) -> np.ndarray:
         """
+        Returns a 3-D Numpy array of dimension (time, component, sample),
+        containing this series' values for all samples.
+
         Parameters
         ----------
         copy
-            Whether to return a copy of the values. Leave it to True unless you know what you are doing.
+            Whether to return a copy of the values, otherwise returns a view.
+            Leave it to True unless you know what you are doing.
 
         Returns
         -------
         numpy.ndarray
-            The values composing the time series guaranteed to be univariate.
+            The values composing the time series.
         """
-        self._assert_univariate()
         if copy:
-            return np.copy(self._df.iloc[:, 0].values)
+            return np.copy(self._xa.values)
         else:
-            return self._df.iloc[:, 0].values
+            return self._xa.values
 
-    def time_index(self) -> pd.DatetimeIndex:
+    def univariate_values(self, copy=True, sample=0) -> np.ndarray:
```

Comment:
Hm again, we could maybe remove the differentiation between univariate and multivariate TimeSeries with a method called get_sample(sample=0) or so. Maybe get_columns(sample=0, columns=None) or so?

Reply:
`univariate_values()` is quite useful e.g. in our metrics methods to fetch values and assert that the series is univariate in one go. I think there's no rush to get rid of it. (good idea for the get_columns() though)